### PR TITLE
Adding fix for #12688 by wrapping M_PI in an #ifndef check

### PR DESCRIPTION
**Pull request by bitbionic**
Added a simple #ifndef check to address issue #12688 - this allows the build to work on a fresh Windows 10 machine with a fresh install of VS 2017

### CODE REVIEW

**Comment by eska014** [comment]
You can change `M_PI` to Godot's `Math_PI`, which is defined in `core/math/math_funcs.h`

**Comment by bitbionic** [comment]
OK made the change

File: drivers/gles3/rasterizer_scene_gles3.cpp [hunk 1]
```diff
@@ -33,11 +33,14 @@
 #include "project_settings.h"
 #include "rasterizer_canvas_gles3.h"
 #include "servers/visual/visual_server_raster.h"
+#include "core/math/math_funcs.h"
 
 #ifndef GLES_OVER_GL
 #define glClearDepth glClearDepthf
 #endif
 
+
```
Comment (akien-mga) [inline drivers/gles3/rasterizer_scene_gles3.cpp]: Please don't add extraneous whitespace, this does not follow the style guide: http://docs.godotengine.org/en/latest/community/contributing/code_style_guidelines.html

File: drivers/gles3/rasterizer_scene_gles3.cpp [hunk 2]
```diff
@@ -33,11 +33,14 @@
 #include "project_settings.h"
 #include "rasterizer_canvas_gles3.h"
 #include "servers/visual/visual_server_raster.h"
+#include "core/math/math_funcs.h"
```
Comment (akien-mga) [inline drivers/gles3/rasterizer_scene_gles3.cpp]: Should be ordered alphabetically as per the style guide.

**Comment by akien-mga** [comment]
Please squash the commits together: http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase

**Comment by ISylvox** [comment]
I'm sorry to say, but I'm difficult to agree with this PR, because I think this PR was made using old master branch a.k.a non up-to the most recent state of upstream/master, which lead into some strange drivers/gles3/rasterizer_scene_gles3.cpp file in bitbionic:12688-M_PI-undeclared branch.

Upstream/Master -- vs -- bitbionic:12688-M_PI-undeclared
1) already has #include "math_funcs.h" -- vs -- there is no #include "math_funcs.h"
2) already has Math_PI (I was the one who changed that M_PI to Math_PI) -- vs -- has M_PI instead of Math_PI (while current upstream/master has Math_PI)

See also:
https://github.com/godotengine/godot/commit/a20ddfdb63ac3f61d3e21acc6042cdcfb1b14b4e

I suggest @bitbionic to compile the most recent and unmodified upstream/master branch to get a clearer issue. :)

**Comment by akien-mga** [comment]
Indeed, your branch is 840 commits behind the current master branch.

**Comment by bitbionic** [comment]
I'm sorry about that. But now I'm confused because I synced my upstream
before all of this and saw gobs of changes come in. Git said my master was
up to date with upstream. I'll be the first to admit I'm new to the
workflow, but thought I had done everything on git correctly. Apparently
not. I'll look at my process and figure out what I did wrong.

**Comment by akien-mga** [comment]
No problem. You can check http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html for tips on the PR workflow we use, including how to sync your repo with the upstream remote.

**Pull request closed without merging**